backend/src/apiserver/storage/experiment_store.go

```diff
@@ -217,6 +236,151 @@ func (s *ExperimentStore) DeleteExperiment(id string) error {
 	return nil
 }
 
+func (s *ExperimentStore) ArchiveExperiment(expId string) error {
+	// ArchiveExperiment results in
+	// 1. The experiment getting archived
+	// 2. All the runs in the experiment getting archived no matter what previous storage state they are in
+	sql, args, err := sq.
+		Update("experiments").
+		SetMap(sq.Eq{
+			"StorageState": api.Experiment_STORAGESTATE_ARCHIVED.String(),
+		}).
+		Where(sq.Eq{"UUID": expId}).
+		ToSql()
+	if err != nil {
+		return util.NewInternalServerError(err,
+			"Failed to create query to archive experiment %s. error: '%v'", expId, err.Error())
+	}
+
+	// TODO(jingzhang36): use inner join to replace nested query for better performance.
+	filteredRunsSql, filteredRunsArgs, err := sq.Select("ResourceUUID").
+		From("resource_references as rf").
+		Where(sq.And{
+			sq.Eq{"rf.ResourceType": common.Run},
+			sq.Eq{"rf.ReferenceUUID": expId},
+			sq.Eq{"rf.ReferenceType": common.Experiment}}).ToSql()
+	if err != nil {
+		return util.NewInternalServerError(err,
+			"Failed to create query to filter the runs in an experiment %s. error: '%v'", expId, err.Error())
+	}
+	updateRunsSql, updateRunsArgs, err := sq.
+		Update("run_details").
+		SetMap(sq.Eq{
+			"StorageState": api.Run_STORAGESTATE_ARCHIVED.String(),
+		}).
+		Where(fmt.Sprintf("UUID in (%s)", filteredRunsSql), filteredRunsArgs...).
+		ToSql()
+	if err != nil {
+		return util.NewInternalServerError(err,
+			"Failed to create query to archive the runs in an experiment %s. error: '%v'", expId, err.Error())
+	}
+
+	updateRunsWithExperimentUUIDSql, updateRunsWithExperimentUUIDArgs, err := sq.
+		Update("run_details").
+		SetMap(sq.Eq{
+			"StorageState": api.Run_STORAGESTATE_ARCHIVED.String(),
+		}).
+		Where(sq.Eq{"ExperimentUUID": expId}).
+		Where(sq.NotEq{"StorageState": api.Run_STORAGESTATE_ARCHIVED.String()}).
+		ToSql()
+	if err != nil {
+		return util.NewInternalServerError(err,
+			"Failed to create query to archive the runs in an experiment %s. error: '%v'", expId, err.Error())
+	}
+
+	// TODO(jingzhang36): use inner join to replace nested query for better performance.
+	filteredJobsSql, filteredJobsArgs, err := sq.Select("ResourceUUID").
+		From("resource_references as rf").
+		Where(sq.And{
+			sq.Eq{"rf.ResourceType": common.Job},
+			sq.Eq{"rf.ReferenceUUID": expId},
+			sq.Eq{"rf.ReferenceType": common.Experiment}}).ToSql()
+	if err != nil {
+		return util.NewInternalServerError(err,
+			"Failed to create query to filter the jobs in an experiment %s. error: '%v'", expId, err.Error())
+	}
+	now := s.time.Now().Unix()
+	updateJobsSql, updateJobsArgs, err := sq.
+		Update("jobs").
+		SetMap(sq.Eq{
+			"Enabled":        false,
+			"UpdatedAtInSec": now}).
+		Where(sq.Eq{"Enabled": true}).
+		Where(fmt.Sprintf("UUID in (%s)", filteredJobsSql), filteredJobsArgs...).
+		ToSql()
+	if err != nil {
+		return util.NewInternalServerError(err,
+			"Failed to create query to archive the jobs in an experiment %s. error: '%v'", expId, err.Error())
+	}
```

Comment:
I forgot to mention...

Did you try this in a real cluster? I believe it won't work.
Because here's how Job and ScheduledWorkflow controller work together:

persistence agent syncs swf CR to kfp api server, if you didn't change the `Enabled` field in swf CRs, persistence agent will soon override the Enabled field back to true again.

Also swf controller only queries swf CRs for scheduling of Jobs, it doesn't know anything about kfp api server's db state.

Therefore, I think the right change is to update the swf CRs to `Enabled: false` one by one. (I'm not sure that can be scalable, so we may need to discuss about the design together first.)

Reply:
For the job part , I relied on backend e2e test to deploy to a real cluster. No manual test through the client directly (the manual test needs to test via client since we don't have the UI part yet)

On the other hand, I think you are right. We'll probably need to use the workflow client's patch method to update job crds. That is to say, we'll need the same behavior as is in EnableJob method, and then we are good. And I added the job crd update part to the archive experiments.